megalinter/MegaLinter.py

```diff
@@ -501,11 +506,14 @@ def list_files_all(self):
         return all_files
 
     def list_git_ignored_files(self):
-        repo = git.Repo(os.path.realpath(self.github_workspace))
-        ignored_files = repo.git.execute("git status --ignored")
-        ignored_files = list(
-            map(lambda x: x + "**" if x.endswith("/") else x, ignored_files)
-        )
+        dirpath = os.path.realpath(self.github_workspace)
+        repo = git.Repo(dirpath)
+        # ignored_files = repo.git.execute(["git", "status", "--ignored"])
+        ignored_files = repo.git.execute(["git", "ls-files", "--exclude-standard", "--ignored", "--others"]).splitlines()
+        ignored_files = map(lambda x: x + "**" if x.endswith("/") else x, ignored_files)
+        # ignored_files will be match againts absolute path (in all_files), so it should be absolute
+        ignored_files = map(lambda x: os.path.join(dirpath, x), ignored_files)
+        ignored_files = sorted(list(ignored_files))
```

Comment:
Any reason for sorting this? You could just return a set from here, might want to do a set comprehension instead of the last map and return that?

Reply:
- sorted to have a reproducible, cross-platform, user-friendly display of the files. (and the display is always done)
- list to be compatible with previous api